crates/fuel-core/src/service/adapters/producer.rs

```diff
@@ -85,3 +106,23 @@ impl fuel_core_producer::ports::Relayer for MaybeRelayerAdapter {
         }
     }
 }
+
+impl fuel_core_producer::ports::BlockProducerDatabase for Database {
+    fn get_block(&self, height: &BlockHeight) -> StorageResult<Cow<CompressedBlock>> {
+        let id = self.get_block_id(height)?.ok_or(not_found!("BlockId"))?;
+        self.storage::<FuelBlocks>()
+            .get(&id)?
+            .ok_or(not_found!(FuelBlocks))
```

Comment:
This should be an option not an error

Reply:
<img width="1187" alt="image" src="https://user-images.githubusercontent.com/18346821/213951968-a8d18324-2ec7-47be-adbd-387b55961aab.png">

We don't process the case of the `Option`. We replace it with the same error on the caller's side. There is not sense to return an `Option` if we don't plan to process it in a different way.

Reply:
I'm sort of ok with this but is does feel like an odd pattern. I guess it's not so bad because the trait is only used by a single crate.